projects/packages/search/tests/php/test-rest-controller.php

```diff
@@ -0,0 +1,274 @@
+
+		$request = new WP_REST_Request( 'GET', '/jetpack/v4/search/plan' );
+		$request->set_header( 'content-type', 'application/json' );
+		$response = $this->server->dispatch( $request );
+		$this->assertEquals( 403, $response->get_status() );
+		$this->assertEquals( 'You are not allowed to perform this action.', $response->get_data()['message'] );
+	}
+
+	/**
+	 * Testing the `POST /jetpack/v4/search/plan` endpoint with editor user.
+	 */
+	public function test_search_plan_successful_authorization() {
+		wp_set_current_user( $this->admin_id );
+
+		$request = new WP_REST_Request( 'GET', '/jetpack/v4/search/plan' );
+		$request->set_header( 'content-type', 'application/json' );
+		$response = $this->server->dispatch( $request );
+		$this->assertEquals( 200, $response->get_status() );
+		$this->assertTrue( $response->get_data()['supports_search'] );
+	}
+
+	/**
+	 * Testing the `POST /jetpack/v4/search/settings` endpoint with editor user.
+	 */
+	public function test_update_search_settings_unauthorized() {
+		wp_set_current_user( $this->editor_id );
+
+		$request = new WP_REST_Request( 'POST', '/jetpack/v4/search/settings' );
+		$request->set_header( 'content-type', 'application/json' );
+		$request->set_body(
+			wp_json_encode(
+				array(
+					'module_active'          => true,
+					'instant_search_enabled' => true,
+				)
+			)
+		);
+		$response = $this->server->dispatch( $request );
+		$this->assertEquals( 403, $response->get_status() );
+		$this->assertEquals( 'You are not allowed to perform this action.', $response->get_data()['message'] );
+	}
+
+	/**
+	 * Testing the `POST /jetpack/v4/search/settings` endpoint with editor user.
+	 */
+	public function test_update_search_settings_success_both_enable() {
+		wp_set_current_user( $this->admin_id );
+		$new_settings = array(
+			'module_active'          => true,
+			'instant_search_enabled' => true,
+		);
+
+		$request = new WP_REST_Request( 'POST', '/jetpack/v4/search/settings' );
+		$request->set_header( 'content-type', 'application/json' );
+		$request->set_body( wp_json_encode( $new_settings ) );
+		$response = $this->server->dispatch( $request );
+		$this->assertEquals( 200, $response->get_status() );
+		$this->assertEquals( $new_settings, $response->get_data() );
+	}
+
+	/**
+	 * Testing the `POST /jetpack/v4/search/settings` endpoint with editor user.
+	 */
+	public function test_update_search_settings_invalid_request_1() {
+		wp_set_current_user( $this->admin_id );
+		$new_settings = array(
+			'module_active'          => false,
+			'instant_search_enabled' => true,
+		);
+
+		$request = new WP_REST_Request( 'POST', '/jetpack/v4/search/settings' );
+		$request->set_header( 'content-type', 'application/json' );
+		$request->set_body( wp_json_encode( $new_settings ) );
+		$response = $this->server->dispatch( $request );
+		$this->assertEquals( 400, $response->get_status() );
+	}
+
+		/**
+		 * Testing the `POST /jetpack/v4/search/settings` endpoint with editor user.
```

Comment:
```suggestion
		 * Testing the `POST /jetpack/v4/search/settings` endpoint with admin user.
```

Also, indentation here needs to be fixed.